tests/spread/extensions/env-injector/task.yaml

```
@@ -0,0 +1,132 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04*
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.env
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  check_env_var() {
+    local snap_app="$1"
+    local env_name="$2"
+    local exp_value="$3"
+
+    local actual_value
+
+    if ! eval "$snap_app" | grep -q "^${env_name}="; then
+      echo "Environment variable '$env_name' is not set."
+      return 1
+    fi
+
+    actual_value=$( "$snap_app" | grep "^${env_name}=" | cut -d'=' -f2-)
+
+    if [ "$actual_value" != "$exp_value" ]; then
+      echo "Environment variable '$env_name' does not match the expected value."
+      echo "Expected: '$env_name=$exp_value', but got: '$env_name=$actual_value'"
+      return 1
+    fi
+
+    return 0 
+  }
+
+  cd "$SNAP_DIR"
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=1 snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the exec-env script is present
+  [ -f squashfs-root/usr/bin/exec-env ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous --classic
+
+  echo "[env-injector] Creating global envfile"
+  touch global.env
+  echo 'HELLO_WORLD="Hello World"' >> global.env
```

```suggestion
  echo 'HELLO_WORLD="Hello World"' >> global.env
```
or 
```suggestion
  echo 'HELLO_WORLD="Hello World"' > global.env
```
?

Also in several other places.

---

I was having some errors about the file not being found, this was before I realize that it was a problem with the snap sandboxing.  :smile: